### PR TITLE
Add affiliation

### PR DESCRIPTION
**Pull request by vyncem**
https://github.com/kaka-ruto/carpitan/issues/25

https://github.com/kaka-ruto/carpitan/assets/3641733/69526ce4-ed67-409e-8973-3f790fc4a2c8

### CODE REVIEW

**Comment by kaka-ruto** [comment]
Thanks for this @vyncem ! I will review soon. Meanwhile, check the failing tests.

File: app/controllers/affiliations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class AffiliationsController < ApplicationController
+  # POST /affiliations or /affiliations.json
+  def create
+    @affiliation = Affiliation.new(politician_id: affiliation_params[:politician_id], relationship: affiliation_params[:relationship], business_id: affiliation_params[:business_attributes][:id]) if affiliation_params[:business_attributes] && affiliation_params[:business_attributes][:id].present?
```
Comment (hmasila) [inline app/controllers/affiliations_controller.rb:4]: You can wrap this in an if-else for easier readability

Also, not sure why we have the two ifs considering an affiliation belongs to a business, therefore, the business_id will always have to exist

File: app/controllers/affiliations_controller.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    @affiliation = Affiliation.new(affiliation_params) if affiliation_params[:business_attributes] && affiliation_params[:business_attributes][:id].blank?
+
+    respond_to do |format|
+      if @affiliation.save
+        format.html { redirect_to business_url(@affiliation), notice: "Business was successfully created." }
```
Comment (hmasila) [inline app/controllers/affiliations_controller.rb:9]: Is the flash message "Business was successfully created." or "Affiliation was successfully created."

File: app/controllers/affiliations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    end
+  end
+
+  private
+    # Only allow a list of trusted parameters through.
+    def affiliation_params
+      params.fetch(:affiliation, {}).permit(:politician_id, :relationship, business_attributes: %i[id name business_phone business_email address industry user_id])
+    end
+end
+# Parameters: {"authenticity_token"=>"[FILTERED]", "affiliation"=>{"politician_id"=>"1", "relationship"=>"owned", "business"=>{"id"=>"1", "name"=>"", "business_phone"=>"", "business_email"=>"", "address"=>"", "industry"=>""}}, "commit"=>"Create Business"}
```
Comment (hmasila) [inline app/controllers/affiliations_controller.rb:24]: You can remove the unused commented out code

**Review by hmasila — CHANGES_REQUESTED** [review]
Looks good. I have left a few comments for you to review/update

**Pull request closed without merging**